### PR TITLE
Add missing methods

### PR DESCRIPTION
**Pull request by cparisineti**
# Description

Please add an informative description that covers that changes made by the pull request and link all relevant issues.

If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.

# All SDK Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes]**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

**Comment by williamzhao87** [comment]
Many Update methods are still missing the without Response methods.

**Comment by cparisineti** [comment]
> Many Update methods are still missing the without Response methods.

Whichs ones @williamzhao87 ?

**Comment by williamzhao87** [comment]
> > Many Update methods are still missing the without Response methods.
> 
> Which ones @williamzhao87 ?
reclassifyJob, updateJob, updateWorker
image

File: sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/JobRouterAsyncClient.java [hunk 1]
```diff
@@ -372,6 +372,150 @@ public Mono<Response<BinaryData>> updateJobWithResponse(String id, BinaryData re
...
+     * }
+     * }</pre>
+     *
+     * @param id The id of the job.
+     * @param resource The resource instance.
+     * @param requestOptions The options to configure the HTTP request before HTTP client sends it.
+     * @return result object.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public Mono<BinaryData> updateJob(String id, BinaryData resource, RequestOptions requestOptions) {
```
Comment (williamzhao87) [inline sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/JobRouterAsyncClient.java:515]: `jobId`

File: sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/JobRouterAdministrationClient.java [hunk 2]
```diff
@@ -1151,6 +1385,82 @@ public Response<BinaryData> updateQueueWithResponse(String id, BinaryData resour
...
+     * }
+     * }</pre>
+     *
+     * @param id The Id of this queue.
+     * @param resource RouterQueue resource.
+     * @param requestOptions The options to configure the HTTP request before HTTP client sends it.
+     * @return a queue that can contain jobs to be routed along with {@link Response}.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public BinaryData updateQueue(String id, BinaryData resource, RequestOptions requestOptions) {
```
Comment (williamzhao87) [inline sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/JobRouterAdministrationClient.java:1454]: `queueId`

File: sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/JobRouterAdministrationClient.java [hunk 3]
```diff
@@ -850,6 +1004,86 @@ public Response<BinaryData> updateExceptionPolicyWithResponse(String id, BinaryD
...
+     * }
+     * }</pre>
+     *
+     * @param id The Id of the exception policy.
+     * @param resource The resource instance.
+     * @param requestOptions The options to configure the HTTP request before HTTP client sends it.
+     * @return result object.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public BinaryData updateExceptionPolicy(String id, BinaryData resource, RequestOptions requestOptions) {
```
Comment (williamzhao87) [inline sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/JobRouterAdministrationClient.java:1083]: `exceptionPolicyId`

File: sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/JobRouterAdministrationClient.java [hunk 4]
```diff
@@ -524,6 +596,88 @@ public Response<BinaryData> updateClassificationPolicyWithResponse(String id, Bi
...
+     * }
+     * }</pre>
+     *
+     * @param id Unique identifier of this policy.
+     * @param resource The resource instance.
+     * @param requestOptions The options to configure the HTTP request before HTTP client sends it.
+     * @return result object.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public BinaryData updateClassificationPolicy(String id, BinaryData resource, RequestOptions requestOptions) {
```
Comment (williamzhao87) [inline sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/JobRouterAdministrationClient.java:677]: `classificationPolicyId`

File: sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/JobRouterAdministrationClient.java [hunk 5]
```diff
@@ -209,6 +209,78 @@ public Response<BinaryData> updateDistributionPolicyWithResponse(String id, Bina
...
+     * }
+     * }</pre>
+     *
+     * @param id The unique identifier of the policy.
+     * @param resource The resource instance.
+     * @param requestOptions The options to configure the HTTP request before HTTP client sends it.
+     * @return result object.
+     */
+    @ServiceMethod(returns = ReturnType.SINGLE)
+    public BinaryData updateDistributionPolicy(String id, BinaryData resource, RequestOptions requestOptions) {
```
Comment (williamzhao87) [inline sdk/communication/azure-communication-jobrouter/src/main/java/com/azure/communication/jobrouter/JobRouterAdministrationClient.java:280]: `distributionPolicyId`

**Comment by williamzhao87** [comment]
Is this method needed?
image

Is this method needed?
image

**Comment by williamzhao87** [comment]
These looks like convenience methods since they take the XXOptions class as parameter.  Shouldn't convenience methods return the entity rather than BinaryData?  Same comment for all convenience methods.
image

**Review by williamzhao87 — CHANGES_REQUESTED** [review]
Some comments still need to be addressed

**Comment by cparisineti** [comment]
> Some comments still need to be addressed

Addressed them. Let me know if there are any more missing. Thank you.

**Comment by williamzhao87** [comment]
> These looks like convenience methods since they take the XXOptions class as parameter. Shouldn't convenience methods return the entity rather than BinaryData? Same comment for all convenience methods.
> image

This is still not addressed

**Comment by williamzhao87** [comment]
> > > Many Update methods are still missing the without Response methods.
> > 
> > 
> > Which ones @williamzhao87 ?
> 
> reclassifyJob, updateJob, updateWorker image

This is still unaddressed

**Comment by williamzhao87** [comment]
Should the XXWithResponse methods also take RequestOptions?
image

**Comment by cparisineti** [comment]
> > > Should the XXWithResponse methods also take RequestOptions? image
> > 
> > 
> > I believe convenience methods do not take RequestOptions. RequestOptions is similar to providing options such as CreateJobOptions. For example protocol methods take RequestOptions in request and return BinaryData.
> 
> Okay then this method should probably also not take RequestOptions:
> 
> public Mono<[RouterWorker](https://apiview.dev/Assemblies/Review/fae6306b94254e02965e844ce05d7ac8?diffRevisionId=0ec28554184f40f18485b108f8fe4f86&diffOnly=False&revisionId=eab4730f3b994bc6a6937311407cfbbf&doc=False#com.azure.communication.jobrouter.models.RouterWorker)> [createWorker](https://apiview.dev/Assemblies/Review/fae6306b94254e02965e844ce05d7ac8?diffRevisionId=0ec28554184f40f18485b108f8fe4f86&diffOnly=False&revisionId=eab4730f3b994bc6a6937311407cfbbf&doc=False#)([CreateWorkerOptions](https://apiview.dev/Assemblies/Review/fae6306b94254e02965e844ce05d7ac8?diffRevisionId=0ec28554184f40f18485b108f8fe4f86&diffOnly=False&revisionId=eab4730f3b994bc6a6937311407cfbbf&doc=False#com.azure.communication.jobrouter.models.CreateWorkerOptions) createWorkerOptions, RequestOptions requestOptions)

Yes, updated it.

**Comment by williamzhao87** [comment]
Looks good to me, once the specs are updated

**Review by williamzhao87 — APPROVED** [review]

**Pull request merged**